### PR TITLE
feat: idempotency message key

### PR DESCRIPTION
**Pull request by valeriangalliat**
This adds a `qkey` option to `sendMessage` to use a custom message key instead of the randomly generated one by default.

This allows using it as idempotency key, plus allows to check whether or not a message already exists in the queue thanks to the new `messageExists` function.

For this to work I had to drop the `id` validation since we can now use anything as message ID instead of the default 32 random chars format.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -187,6 +192,14 @@ declare namespace RedisSMQ {
 		 */
 		message: string;
 
+		/**
+		 * Key for the message to deduplicate unique messages
+		 *
+		 * @type {string}
+		 * @memberof SendMessageOptions
+		 */
+		qkey?: string;
```
Comment (alexbouchardd) [inline index.d.ts:201]: Any reason for labelling the key `qkey` the `q` just makes it more confusing?

**Review by alexbouchardd — APPROVED** [review]
Sweeeet! Should we bother with proposing upstream changes?

**Comment by valeriangalliat** [comment]
> Sweeeet! Should we bother with proposing upstream changes?

It's very specific to the way we use this library and doesn't sound like something one would want out of a generic Pub/Sub queue, so I wouldn't propose it upstream

File: index.d.ts (hunk 1 above)
Comment (valeriangalliat) [reply index.d.ts:201]: Consistency with the `qname` option is the reason!

**Comment by valeriangalliat** [comment]
@alexbouchardd Can you bump the version and publish it on npm? 🙏

**Pull request merged**